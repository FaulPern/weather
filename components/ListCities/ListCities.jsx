import Link from 'next/link'
import WeatherIcons from '../WeatherIcons/WeatherIcons'

const ListCities = props => {
  const cities = { ...props.data }
  const date = new Date()
  const day = date.getDay()
  return (
    <ul className='grid'>
      {Object.entries(cities).map(city => (
        <li key={city.toString()}>
          <Link href={`/weather/${city[1].slug}`}><h2>{city[1].name}</h2></Link>
          <WeatherIcons name='sun' />
          <span>{city[1].forecast[day].tempmin} / {city[1].forecast[day].tempmax}</span>
        </li>
      ))}
    </ul>
  )
}

export default ListCities
