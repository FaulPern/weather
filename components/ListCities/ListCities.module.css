.listCities {
  list-style: none;
}

.icon {
  width: 30px;
}