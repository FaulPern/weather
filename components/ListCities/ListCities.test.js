import { mount } from 'enzyme'
import ListCities from './ListCities'

describe('ListCities component', () => {
  const mountComponent = mount(<ListCities />)
  it('should render', () => {
    expect(mountComponent).toMatchSnapshot()
  })
})
