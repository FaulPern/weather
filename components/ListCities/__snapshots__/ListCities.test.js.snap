// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListCities component should render 1`] = `
<ListCities>
  <ul
    className="grid"
  />
</ListCities>
`;
